### PR TITLE
Add theme (`-t`/`--theme=NAME_OR_ID`) parameter to `theme push`/`theme pull` commands

### PR DESCRIPTION
**Pull request by karreiro**
### WHY are these changes introduced?

Users can't set the theme name when running commands like `shopify theme push -u` (https://github.com/Shopify/shopify-cli/issues/1431).

### WHAT is this pull request doing?

This PR introduces a new parameter representing the theme name `-t`/`--theme=NAME_OR_ID`. So, now users can perform operations `shopify theme push -u -t my_theme`.

For consistency, this PR also:
- Adds the `-t`/`--theme=NAME_OR_ID` to `theme pull`
- Adds the `-t`/`--theme=NAME_OR_ID` to `theme push` (globally, even without the`--unpublished` parameter)

Thus, now users may rely on the `-t <NAME_OR_ID>` to `push`/`pull` a theme, instead of only relying on `-i <ID>`.

### How to test your changes?

Run the following commands for checking the impact of the new parameter on `theme push`:

```bash
shopify theme push --live
shopify-dev theme push --theme abc
shopify-dev theme push -t abc
shopify-dev theme push -u -t abc
```

Run the following commands for checking the impact of the new parameter on `theme pull`:
```bash
shopify-dev theme pull --theme abc
shopify-dev theme pull -t abc
```

### Post-release steps
Update documentation with the new `-t`/`--theme=THEME` parameter: https://shopify.dev/themes/tools/cli/theme-commands#push, https://shopify.dev/themes/tools/cli/theme-commands#pull.

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [x] I've included any post-release steps in the section above.

### CODE REVIEW

**Comment by karreiro** [comment]
Hi, @MeredithCastile! Thanks again for the thoughts you've put into this and for the questions you've raised. @macournoyer and I talk a bit about this PR in the latest theme meeting.

Considering some extra context for this use case:
- the flag `--themename` is not intuitive
- user's PR is not in the scope theme CLI lifecycle in this context
- repurposing/renaming the flag `--nodelete` to free up `-n` could be dangerous as users unaware of the change could lose files
- keeping the paradigm of `command --flags` is valuable to keep the DX consistent and not break users expectation

We've reasoned that **bundling the flags is a great idea** 🚀

Thus, users could now rely on the `-t`/`--theme` flag to pass the **theme id** or the **theme name**. What do you think about this approach?

Thanks! :)

**Comment by MeredithCastile** [comment]
@karreiro Seems like a good solution! Thanks. 🙌

Please do make sure that the --help and docs are updated to reflect these changes. If you run into any problems updating [the command](https://shopify.dev/themes/tools/cli/theme-commands#optional-parameters) in .dev docs, I'd recommend reaching out to @shainaraskas.

File: lib/project_types/theme/messages/messages.rb [hunk 1]
```diff
@@ -57,6 +57,7 @@ module Messages
               Usage: {{command:%s theme push [ ROOT ]}}
 
               Options:
+                {{command:-t, --theme=THEME}}     Theme ID or name of the remote theme.
```
Comment (macournoyer) [inline lib/project_types/theme/messages/messages.rb:60]:
```suggestion
                {{command:-t, --theme=NAME_OR_ID}}     Theme ID or name of the remote theme.
```

Seems clearer to me.

File: lib/project_types/theme/messages/messages.rb [hunk 2]
```diff
@@ -57,6 +57,7 @@ module Messages
...
                 {{command:-i, --themeid=THEMEID}} Theme ID. Must be an existing theme on your store.
```
Comment (macournoyer) [inline lib/project_types/theme/messages/messages.rb:61]: Any reason, other than backward compat, to keep this arg now that we have `-t`?

If it's just for backward compatibility, maybe we should deprecate `-i` and remove it from the usage doc?

**Review by macournoyer — APPROVED** [review]
I think it's confusing to have two args do the same thing now :/

`-i ID` and `-t ID`

But looks good on the code side.

File: lib/project_types/theme/messages/messages.rb (hunk 1 above)
Comment (karreiro) [reply lib/project_types/theme/messages/messages.rb]: Great suggestion! Fixed :)

File: lib/project_types/theme/messages/messages.rb (hunk 2 above)
Comment (karreiro) [reply lib/project_types/theme/messages/messages.rb]: Excellent suggestion! Fixed :)

**Comment by macournoyer** [comment]
> Does it sound like a good plan? :)

YES! 🙏

**Pull request merged**